toolbox

```diff
@@ -69,16 +69,20 @@ run() {
         echo "Setting up user '${USER_NAME}' (with 'sudo' access) inside the container..."
         echo "(NOTE that, if 'sudo' and related packages are not present in the image already,"
         echo "this may take some time. But this will only happen now that the toolbox is being created)"
-        cat <<EOF > /tmp/${TOOLBOX_NAME}-user-setup.sh
+	local tmp_user_setup=`mktemp /tmp/${TOOLBOX_NAME}-user-setup-XXXXXX.sh`
```

Comment:
Is the "tmp_user_setup" script created always with root permissions, or also by normal users? In the first case it could be an option to use /run for this.
But please use ${TMPDIR} and don't hardcode /tmp. So if TMPDIR is not set, set it to /tmp, else allow to override this location.

Reply:
It (tries to) installs packages and fiddle with users and groups, so it's always the root user inside the container that has to run it. In theory, I can create it as a user, but only if the toolbox is started with the `-u` option, otherwise there is no user besides root inside it.

So, I think this is basically a "yes, it is always created by root and needs to be executed by root. Does this mean I should put it in /run then?

Reply:
The file is created outside the container. Is the script always running with root rights? Means can it write to a location owned by root? Your answer doesn't sound like that. So I would go for ${TMPDIR} pointing initially to /tmp.

Reply:
Ah, yes, the script is created outside and copied inside. it's created by whoever runs `toolbox`, which indeed may be a normal user, not necessarily having access to a root-owned path. I'm ok with ${TMPDIR}. Maybe I could even go for ${HOME}/${TOOLBOX_NAME}-user-setup-XXXXXX.sh , as I expect whoever s trying to create a toolbox to have a real home directory? Or are there use-cases where this may not be true?

Reply:
Yes, there are accounts with no writeable home directory. On the other side, it's unlikely they will run the toolbox command.
${HOME}/.${TOOLBOX_NAME}-user-setup-XXXXXX.sh should be fine (I would use the "." as nearly everything creating temporary files in the home directory)

Reply:
Exactly my point! A system user without (or with a special) homedir, is very unlikely to use `toolbox -u`. Also, totally agree on leading '.' in the name